### PR TITLE
Methods to downgrade from 2020 -> 2018 / 2015

### PR DESCRIPTION
**Pull request by MadhuvanthG**
- Unlike the existing downgrade/upgrade functions which are part of a separate package, for 2020 version structs, these schema upgrade/downgrade are methods on the types itself

- Because 2020 spec only has additions, used a simple approach of unmarshaling the  2020 object directly into 2018, which would get rid of the additional elements in 2020. But with this approach, (2020)ViewingPolicy.Allocation gets unmarshaled  into (2018)ViewingPolicy.ActionProperty. So explicitly removing it in the downgraded version

- **Will add unit tests**, but want to first gather feedback on the approach

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
+Details on SCTE 224 can be found at: https://www.scte.org/standards-development/library/standards-catalog/ansiscte-224-2018r1/
 
-The most recent spec: https://www.scte.org/SCTEDocs/Standards/ANSI_SCTE%20224%202018r1.pdf
+2015 schema: http://www.scte.org/schemas/224/SCTE224-20151115.xsd (link no longer working, refer to the spec located alongside structs in this repo)
 
-2015 schema: http://www.scte.org/schemas/224/SCTE224-20151115.xsd
+2018 schema: http://www.scte.org/schemas/224/SCTE224-20180501.xsd (link no longer working, refer to the spec located alongside structs in this repo)
 
-2018 schema: http://www.scte.org/schemas/224/SCTE224-20180501.xsd
+Latest schema (currently points to 2020): https://www.scte.org/standards-development/library/standards-catalog/ansiscte-224-2018r1/
```
Comment (superflaco) [inline README.md:11]: wow, super weird that the latest schema link https://scte-cms-resource-storage.s3.amazonaws.com/SCTE_224_20200407-1600425639358.xsd is found on a page referencing the 2021 schema located on a url mentioning 2018r1 @skurkows seems like SCTE still has wacky page structures

File: types/scte224v20200407/sctestructures.go [hunk 2]
```diff
@@ -42,6 +46,46 @@ type Media struct {
...
 
+func (m *Media) Get2018() (*scte224_2018.Media, error) {
+	encodedMedia, err := xml.Marshal(m)
+	if err != nil {
+		return nil, fmt.Errorf("error attempting to downgrade the 2020 element, %+v", err)
+	}
+
+	// Unmarshal into 2018 struct
+	var media2018 *scte224_2018.Media
+	err = xml.Unmarshal(encodedMedia, &media2018)
```
Comment (superflaco) [inline types/scte224v20200407/sctestructures.go:57]: how does this benchmark?  I think I avoided a marshal-unmarshal route with 2015-2018 conversions both due to the namespace difference as well as thinking marshalling performance would be terrible.  Might not be a big deal though mostly curious.

File: types/scte224v20200407/viewingpolicy.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	"fmt"
+	"time"
+
+	"github.com/Comcast/scte224structs/convert"
+	scte224_2015 "github.com/Comcast/scte224structs/types/scte224v20151115"
+	scte224_2018 "github.com/Comcast/scte224structs/types/scte224v20180501"
+)
+
+//Table 12
+type ViewingPolicy struct {
```
Comment (superflaco) [inline types/scte224v20200407/viewingpolicy.go:14]: got too cumbersome in one big file?

**Review by superflaco — APPROVED** [review]
Seems fine, but I just get squeamish about all the marshalling roundtrips killing performance which may be overthinking it.  Let's get the readme cleaned up before committing though.

File: types/scte224v20200407/sctestructures.go (hunk 2 above)
Comment (MadhuvanthG) [reply types/scte224v20200407/sctestructures.go:57]: Valid concern with Unmarshaling/marshaling. I am skeptical about it too. 

Yet to benchmark, was going to do so while adding unit tests.

We could leave the PR open, until I benchmark and observe results.

File: types/scte224v20200407/viewingpolicy.go (hunk 3 above)
Comment (MadhuvanthG) [reply types/scte224v20200407/viewingpolicy.go:14]: Yes, was getting bigger with the new methods

File: types/scte224v20200407/sctestructures.go (hunk 2 above)
Comment (MadhuvanthG) [reply types/scte224v20200407/sctestructures.go:57]: We still have to dereference each of the values, if we were to manually copy from one struct to another. 

So I thought, at least theoretically, there wouldn’t be too much difference between marshaling/unmarshaling approach and copying approach.

**Comment by MadhuvanthG** [comment]
> Seems fine, but I just get squeamish about all the marshalling roundtrips killing performance which may be overthinking it. Let's get the readme cleaned up before committing though.

Marshaling/Unmarshaling wasn't helping performance at all. I did some benchmarks and for a Media with ~1000 media points:
1. xml marshal/unmarshal approach took ~400 milliseconds
2. field-by-field copy took under 5 milliseconds

Although approach (1) leads to simpler code, that's a huge performance penalty to pay. So, changed the methods to use approach (2) and added enough unit tests to ensure downgrade of different structs are working as expected.

**Pull request merged**